frontend/TestRun/ResultsGraph.svelte

```diff
@@ -94,56 +108,145 @@
         graph.options.lazy = true;
         graph.options.plugins.tooltip = {
+            enabled: false,
+            mode: "point",
+            intersect: true,
             position: "above",
-            usePointStyle: true,
-            bodyFont: {
-                size: 16,
-                family: 'sans-serif',
+            external: function (context) {
+                if (activeTooltip) {
+                    return;
+                }
+                let tooltipEl = document.getElementById("chartjs-tooltip");
+                if (!tooltipEl) {
+                    tooltipEl = document.createElement("div");
+                    tooltipEl.id = "chartjs-tooltip";
+                    tooltipEl.innerHTML = "<div></div>";
+                    document.body.appendChild(tooltipEl);
+                }
+
+                const tooltipModel = context.tooltip;
+                if (tooltipModel.opacity === 0 && !activeTooltip) {
+                    tooltipEl.remove();
+                    return;
+                }
+
+                if (tooltipModel.body) {
+                    const titleLines = tooltipModel.title || [];
+                    const bodyLines = tooltipModel.body.map((b) => b.lines);
+
+                    let innerHtml = '<div class="custom-tooltip">';
+
+                    tooltipModel.dataPoints.forEach((dataPoint, i) => {
+                        const colors = tooltipModel.labelColors[i];
+                        const pointStyle = tooltipModel.labelPointStyles[i];
+
+                        const span = document.createElement("span");
+                        span.style.display = "inline-block";
+                        span.style.width = "10px";
+                        span.style.height = "10px";
+                        span.style.marginRight = "8px";
+                        span.style.borderRadius = "50%";
+                        span.style.backgroundColor = colors.backgroundColor;
+                        span.style.border = `2px solid ${colors.borderColor}`;
+
+                        // Add title
+                        innerHtml += `<div class="tooltip-title">`;
+                        innerHtml += span.outerHTML;
+                        innerHtml += titleLines[i];
+                        innerHtml += `</div>`;
+
+                        // Add corresponding changes
+                        const changes = tooltipModel.body[i]?.lines || [];
+                        changes.forEach((line) => {
+                            innerHtml += `<div class="tooltip-body">${line}</div>`;
+                        });
+
+                        // Add Open button for this point
+                        const runId = context.chart.data.datasets[dataPoint.datasetIndex].data[dataPoint.dataIndex].id;
+                        innerHtml += `<button class="tooltip-button" onclick="document.getElementById('graph-${test_id}-${index}').dispatchEvent(new CustomEvent('openRun', {detail: '${runId}'}))">Open</button>`;
```

Comment:
Nitpick: can we make this button small, and a bit less taking focus?

Reply:
how about this?
![image](https://github.com/user-attachments/assets/d829cfd5-1ea4-409f-9cd2-15a52cb97f24)

Reply:
It's better, maybe it should go on the right side ? maybe the expend icon ?

Reply:
external link button added and moved it to right:
![image](https://github.com/user-attachments/assets/b2c94f94-89e5-43bd-b3f6-f7326d86b2af)